test/zksync-tests/AllowanceTransferInvariants.test.ts

```diff
@@ -0,0 +1,105 @@
+
+            let permit: PermitSingle = {
+                details: {
+                    token: token.address,
+                    amount: permitted,
+                    expiration: defaultExpiration,
+                    nonce: defaultNonce,
+                },
+                spender: choosedSpender.address,
+                sigDeadline: 2000000000,
+            };
+
+            let startBalanceFrom: BigNumberish = await token.connect(choosedPermitter).balanceOf(choosedPermitter.address);
+
+            const sign: Uint8Array = getCompactPermitSignature(permit, choosedPermitter.privateKey, await permit2.DOMAIN_SEPARATOR());
+
+            await (await permit2.connect(choosedPermitter)["permit(address,((address,uint160,uint48,uint48),address,uint256),bytes)"](choosedPermitter.address, permit, sign)).wait();
+
+            await (await permit2.connect(choosedSpender)["transferFrom(address,address,uint160,address)"](choosedPermitter.address, choosedSpender.address, permitted, token.address)).wait();
+
+            expect(await token.connect(choosedSpender).balanceOf(choosedSpender.address)).to.be.equal(permitted);
+            expect(await token.connect(choosedPermitter).balanceOf(choosedPermitter.address)).to.be.equal(startBalanceFrom.sub(permitted));
+        });
+    });
+
+
+    describe("Balance Equals Spent", function () {
+        it('balance should equals spent', async function () {
+            let permitted: BigNumber = DECIMAL_MULT;
+
+            let permit: PermitSingle = {
+                details: {
+                    token: token.address,
+                    amount: permitted,
+                    expiration: defaultExpiration,
+                    nonce: defaultNonce,
+                },
+                spender: choosedSpender.address,
+                sigDeadline: 2000000000,
+            };
+
+            let startBalanceFrom: BigNumberish = await token.connect(choosedPermitter).balanceOf(choosedPermitter.address);
+
+            const sign: Uint8Array = getCompactPermitSignature(permit, choosedPermitter.privateKey, await permit2.DOMAIN_SEPARATOR());
+
+            await (await permit2.connect(choosedPermitter)["permit(address,((address,uint160,uint48,uint48),address,uint256),bytes)"](choosedPermitter.address, permit, sign)).wait();
+
+            await (await permit2.connect(choosedSpender)["transferFrom(address,address,uint160,address)"](choosedPermitter.address, choosedSpender.address, permitted, token.address)).wait();
+
+            expect(await token.connect(choosedSpender).balanceOf(choosedSpender.address)).to.be.equal(startBalanceFrom.sub(await token.connect(choosedPermitter).balanceOf(choosedPermitter.address)));
+        });
+    });
+
+
+    describe("Permit2 Never Holds Balance", function () {
+        it('permit2 should have zero balance', async function () {
+            expect(await token.connect(spender1).balanceOf(permit2.address)).to.be.equal(ethers.constants.Zero);
+        });
+    });
+
+
+});
```

Comment:
New line

Reply:
I apologize, I will definitely review the configuration of my linter.